File: Data Structures & Algorithms/lemonade-change/submission-2.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        count ={}
        for elem in bills:
            if elem == 5:
                count[elem]=1+count.get(elem,0)
            elif elem == 10:
                if count.get(5,0)>=1:
                    count[5]-=1
                    count[elem]=1+count.get(elem,0)
                else:
                    return False
            elif elem == 20:
                if count.get(5,0)>=3:
                    count[5]-=3
                    count[elem]=1+count.get(elem,0)
                elif count.get(5,0)>=1 and count.get(10,0)>=1:
                    count[5]-=1
                    count[10]-=1
                    count[elem]=1+count.get(elem,0)
                else:
                    return False
        return True

            

File: Data Structures & Algorithms/boats-to-save-people/submission-2.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int l = 0;
        Arrays.sort(people);
        int r = people.length - 1;
        int count = 0;
        while(l<=r){
            int left = limit - people[r];
            r--;
            count ++;
            if(l<=r && left>=people[l]){
                
                l++;
            }
        }
        return count;
        
    }
}